Tasks/AzureKeyVaultV1/operations/azure-arm-keyvault.ts

```diff
@@ -28,21 +29,45 @@ export class KeyVaultClient extends azureServiceClient.ServiceClient {
     }
 
     public async invokeRequest(request: webClient.WebRequest): Promise<webClient.WebResponse> {
-        try {
-            var response = await this.beginRequest(request);
-            if (response.statusCode == 401) {
-                var vaultResourceId = this.getValidVaultResourceId(response);
-                if(!!vaultResourceId) {
-                    console.log(tl.loc("RetryingWithVaultResourceIdFromResponse", vaultResourceId));
-                    
-                    this.getCredentials().activeDirectoryResourceId = vaultResourceId; // update vault resource Id
-                    this.getCredentials().getToken(true); // Refresh authorization token in cache
-                    var response = await this.beginRequest(request);
+        const maxRetryCount: number = 5;
+        const retryIntervalInSeconds: number = 2;
+        const retriableErrorCodes = ["ETIMEDOUT", "ECONNRESET", "ENOTFOUND", "ESOCKETTIMEDOUT", "ECONNREFUSED", "EHOSTUNREACH", "EPIPE", "EA_AGAIN", "EAI_AGAIN"];
+        const retriableStatusCodes = [408, 409, 500, 502, 503, 504];
+        let timeToWait: number = retryIntervalInSeconds;
+        let retryCount: number = 0;
+        
+        while(true) {
```

Comment:
Have a max number of retries maybe?

Reply:
`while(++retryCount < maxRetryCount)` instead maybe

Reply:
retry logic is inside the loop